### ISSUE TITLE
Configuration page http_server should treat URLs as case-insensitive

### ISSUE THREAD

**Issue by schanck** [anxdpanic/plugin.video.youtube#394]
### Context

* Add-on Version: 6.8.24+matrix.1
* Kodi Version: 20.0
* Kodi GUI Language: English
* Operating System: Lubuntu 22.10
* Operating System Language: English
* Skin: Aeon Tajo (skin dependent, but multiple skins are affected)

------

### Expected Behavior

`http://kodi_IP:50152/API` should load the configuration page.
I.e. it should behave the same as `http://kodi_IP:50152/api`

This is needed because some skins uppercase the server address to ALL CAPS, so you are told to go to a URL ending in /API , which doesn't work. It could be argued that this is a bug in the skins. However, I think it's more appropriate, and fairly trivial (see proposed fix at bottom) to fix the issue in this plugin. Otherwise, a user friendly experience relies on every skin developer never uppercasing dialog strings, which is already known to not be the case.

Skins which are affected by this (i.e. that uppercase the server address) include:
* Aeon Tajo
* Amber
* Metropolis
* Mimic-LR

I tested all the ones available by default on Kodi 20.0, except for Bello 8 which wouldn't install due to a dependency issue.

------

### Current Behavior

A confusing 501 error is shown:
```
Error response

Error code: 501

Message: Not Implemented.

Error code explanation: 501 - Server does not support this operation.
```

Ideally, a more helpful message than this would be shown in response to unrecognised URLs, perhaps even a "click here to go to the config page that you were probably trying to reach." However, I realise this may be a limitation of the underlying BaseHTTPServer, and that this is getting into discussion of a possible different bug / feature request.

------

### Steps to Reproduce

1. Have a skin applied that uppercases strings in this plugin's dialogs, such as **Aeon Tajo**
2. Enable the `YouTube Add-on API Configuration` page
3. Go to the server URL which is shown of `HTTP://KODI_IP:50152/API`
4. Get 501 error instead of expected config page

------

### Proposed fix (completely untested)

Every comparison to `stripped_path` is done against a lowercase string, so I think `stripped_path` should be normalised to lower case upon initialisation, [which it currently isn't](https://github.com/anxdpanic/plugin.video.youtube/blob/85104cc22f53806b1f8ed519bc691177151d8303/resources/lib/youtube_plugin/kodion/utils/http_server.py#L71)
        `stripped_path = self.path.rstrip('/').lower()`

[The existing logic should then work as expected with URLs ending in /API](https://github.com/anxdpanic/plugin.video.youtube/blob/85104cc22f53806b1f8ed519bc691177151d8303/resources/lib/youtube_plugin/kodion/utils/http_server.py#L104)
`            elif api_config_enabled and stripped_path == '/api':`

 
** **
** **

**Comment by schanck** [comment]
Updated issue: this is now known to happen with at least 4 skins that are available by default

**Comment by schanck** [comment]
Closing as I see this was [resolved in v6.8.25](https://github.com/anxdpanic/plugin.video.youtube/releases/tag/v6.8.25), thanks!

`            elif api_config_enabled and stripped_path.lower() == '/api':`
